src/Trustline.ts

```diff
@@ -3,119 +3,205 @@ import { Utils } from './Utils'
-   * @param decimals (optional) number of decimals can be provided manually
-   * @param gasLimit (optional)
-   * @param gasPrice (optional)
+   * Prepares a tx object for creating a trustline update request. Called by initiator
+   * of update request.
+   * @param network Address of a currency network.
+   * @param counterparty Address of counterparty who receives trustline update request.
+   * @param given Proposed creditline limit given by iniator to counterparty,
+   *              i.e. 1.23 if network has to 2 decimals.
+   * @param received Proposed creditline limit received by iniator from counterparty,
+   *                 i.e. 1.23 if network has to 2 decimals.
+   * @param options Options for creating tx. See type `TLOptions` for more information.
+   * @param options.decimals Decimals of currency network can be provided manually if known.
+   * @param options.gasLimit Custom gas limit.
+   * @param options.gasPrice Custom gas price.
    */
-  public prepareUpdate (
+  public async prepareUpdate (
     network: string,
-    debtor: string,
-    creditLineGiven: number,
-    creditLineReceived: number,
-    { decimals, gasLimit, gasPrice }: TLOptions = {}
-  ): Promise<any> {
-    const { currencyNetwork, transaction, user, utils } = this
-    const { calcRaw } = utils
-    return currencyNetwork.getDecimals(network, decimals)
-      .then(dec => transaction.prepFuncTx(
-        user.address,
+    counterparty: string,
+    given: number | string,
+    received: number | string,
+    options: TLOptions = {}
+  ): Promise<TxObject> {
+    try {
+      const { _currencyNetwork, _transaction, _user, _utils } = this
+      let { decimals, gasLimit, gasPrice } = options
+      decimals = await _currencyNetwork.getDecimals(network, decimals)
+      return _transaction.prepFuncTx(
+        _user.address,
         network,
         'CurrencyNetwork',
         'updateTrustline',
-        [ debtor, calcRaw(creditLineGiven, dec), calcRaw(creditLineReceived, dec) ],
+        [ counterparty, _utils.calcRaw(given, decimals), _utils.calcRaw(received, decimals) ],
         { gasPrice, gasLimit }
-      ))
+      )
+    } catch (error) {
+      return Promise.reject(error)
+    }
   }
 
   /**
-   * Prepares a raw transaction for accepting a trustline update request
-   * @param network address of currency network
-   * @param creditor address of counterparty who sent trustline udpate request
-   * @param creditLineGiven value of credit line given to counterparty, i.e. 1.23 if network has to 2 decimals
-   * @param creditLineReceived value of credit line received by counterparty, i.e. 1.23 if network has to 2 decimals
-   * @param decimals (optional) number of decimals can be provided manually
-   * @param gasLimit (optional)
-   * @param gasPrice (optional)
+   * Prepares a tx object for accepting a trustline update request. Called
+   * by receiver of initial update request.
+   * @param network Address of a currency network.
```

Comment:
networkAddress

Reply:
@berndbohmeier This is not relevant for this PR. Issue was to integrate a doc tool. The doc-block matches the method perfectly, a different name would be part of a refactoring tough

Reply:
true, but as in this PR was also some refactoring done I mentioned it.

Reply:
But you are right, we can make another issue out of this.

Reply:
https://github.com/trustlines-network/clientlib/issues/87